src/Common/src/Microsoft/Win32/SafeHandles/SafePerfProviderHandle.cs

```diff
@@ -0,0 +1,28 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.Threading;
+
+namespace Microsoft.Win32.SafeHandles
+{
+    internal sealed class SafePerfProviderHandle : SafeHandleZeroOrMinusOneIsInvalid
+    {
+        private SafePerfProviderHandle() : base(true) { }
+
+        protected override bool ReleaseHandle()
+        {
+            IntPtr tempProviderHandle = handle;
+
+            if (Interlocked.Exchange(ref handle, IntPtr.Zero) != IntPtr.Zero)
```

Comment:
I don't think this is necessary- I believe the runtime only lets this method get hit once. @jkotas, am I reading the implementation correctly?

Reply:
Correct, this should not be necessary.

Reply:
fixed